src/anyp/UriScheme.cc

```diff
@@ -48,6 +48,25 @@ AnyP::UriScheme::Init()
     }
 }
 
+const AnyP::ProtocolType
+AnyP::UriScheme::FindProtocolType(const SBuf &str)
```

Comment:
Consider using an SBuf parameter so that you can modify it w/o explicit copying. In any case, please rename `str` to `scheme`.

```suggestion
AnyP::UriScheme::FindProtocolType(SBuf scheme)
```

Reply:
Rename done. The parameter type change would just move the copy into all callers in order to be able to pass non-const parameter. This way we have only one copy ensure it is always correctly done.